src/main/webapp/consentGroup/NewConsentGroup.js

```diff
@@ -134,7 +134,9 @@ class NewConsentGroup extends Component {
   uploadFiles = (projectKey) => {
     Files.upload(this.props.attachDocumentsURL, this.state.files, projectKey, this.state.user.displayName, this.state.user.userName, true)
       .then(resp => {
-        window.location.href = this.getRedirectUrl();
+        Project.getProjectType(this.props.serverURL, this.props.projectKey).then(type => {
+          window.location.href =  [this.props.serverURL, type.data.projectType, "show", this.props.projectKey, "?tab=consent-groups"].join("/");
+        });
```

Comment:
Ouch ... I'm pretty sure I missed this in previous reviews. We should not be reloading the page in react components. These should trigger a react lifecycle event instead if we think we need to reload page data. State should drive what happens in the view here. Ideally, we'd fire off an async post, then update state with the results.

Reply:
I understand your concern here, but this method only intends to redirect to preview page after creation, it's not reloading the same page.

On the other hand, it's  prefered to use React Routes, which could be used after we have a Higher order Component making the link between Grails and React, and containing a Switch to redirect each path with its props.

Either way i'll make a change here adding Async/Await in order to prevent this redirection inside a promise.

Reply:
OK, I see what you're trying to do here. The effect of `window.location.href` is to reload the page with the destination. In react apps, we should never have to set the window's location. In our world, we are transitioning from older style gsps, so we might need to redirect, but we absolutely need to make sure this is temporary code. 

We need two things here, a comment explaining what we're doing and a link to a tech-debt ticket that covers fixing this. With those two things, 👍

Reply:
Here's the tech-debt story.
[BTRX-637](https://broadinstitute.atlassian.net/browse/BTRX-637) : ORSP - Technical debt - improve react redirections